src/aws-util/Cargo.toml

```diff
@@ -10,16 +10,27 @@ publish = false
 workspace = true
 
 [dependencies]
+anyhow = "1.0.66"
 aws-config = { version = "1.1.1", default-features = false }
 aws-sdk-s3 = { version = "1.8.0", default-features = false, features = ["rt-tokio"], optional = true }
 aws-smithy-runtime-api = "1.1.1"
 aws-smithy-runtime = { version = "1.1.1", features = ["connector-hyper-0-14-x"] }
 aws-types = "1.1.1"
+bytes = "1.3.0"
+bytesize = "1.1.0"
+http = "0.2.8"
 hyper-tls = { version = "0.5.0" }
+mz-ore = { path = "../ore", default-features = false, features = ["metrics", "async", "bytes_", "region"] }
+thiserror = "1.0.37"
+tokio = { version = "1.32.0", default-features = false, features = ["fs", "macros", "sync", "rt", "rt-multi-thread"] }
+uuid = { version = "1.7.0", features = ["v4"] }
 workspace-hack = { version = "0.0.0", path = "../workspace-hack", optional = true }
 
+[dev-dependencies]
+mz-ore = { path = "../ore", default-features = false, features = ["test"] }
+
 [features]
-default = ["workspace-hack"]
+default = ["workspace-hack", "s3"]
```

Comment:
Had to add `s3` in default, otherwise the test was not running.

Reply:
We'll want to find a way to avoid this so that we don't force the `s3` dependency on crates that don't need it. I think possibly the thing that's missing is passing `--all-features` to the Cargo test runner in CI?

```
diff --git a/ci/test/cargo-test/mzcompose.py b/ci/test/cargo-test/mzcompose.py
index 64355a811e..44d1a4a116 100644
--- a/ci/test/cargo-test/mzcompose.py
+++ b/ci/test/cargo-test/mzcompose.py
@@ -89,6 +89,7 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
             "cargo",
             "llvm-cov",
             "nextest",
+            "--all-features",
             "--release",
             "--no-clean",
             "--workspace",
@@ -137,6 +138,7 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
                     "cargo",
                     "nextest",
                     "run",
+                    "--all-features",
                     "--profile=ci",
                     f"--partition=count:{partition}/{total}",
                     # Most tests don't use 100% of a CPU core, so run two tests per CPU.
```

Reply:
I see, I'll do that

Reply:
@MaterializeInc/testing can probably help too!

Reply:
Definitely. @moulimukherjee Should I push into this PR directly?

Reply:
@def- Sure, feel free to do so 🙇‍♀️